### PR TITLE
Remaining nonflask db updates

### PR DESCRIPTION
**Pull request by bsweger**
This PR updates some of the previous changes I'd made to get rid of the database interfaces from non-Flask code. The changes are based on team discussions and code reviews and reflect our latest thinking on how best to handle database access within the broker:

* All db access is not through a flask application context (i.e., no more stand-alone database session context)
* When creating a one-off flask context, do it with the validator's app.py instead of the broker's app.py, which is more resource intensive
* Move a problematic interface function to `function_bag.py` and re-think its session handling in an attempt to fix the intermittent SQLAlchemy errors on local integration test runs

### CODE REVIEW

File: dataactcore/interfaces/function_bag.py [hunk 1]
```diff
@@ -118,3 +119,125 @@ def checkNumberOfErrorsByJobId(jobId, errorType='fatal'):
...
+    else:
+        # create a new record validation job and add dependencies if necessary
+        if fileType == "awardee_attributes":
+            d1ValJob = sess.query(Job).\
+                filter(Job.submission_id == submissionId,
+                       Job.file_type_id == FILE_TYPE_DICT['award_procurement'],
+                       Job.job_type_id == JOB_TYPE_DICT['csv_record_validation']).\
+                first()
+            if d1ValJob is None:
+                raise Exception("Cannot create E job without a D1 job")
```
Comment (cmc333333) [inline dataactcore/interfaces/function_bag.py]: Hrm. Note for later: E relies on D1 _and_ D2.

File: dataactcore/interfaces/function_bag.py [hunk 2]
```diff
@@ -118,3 +119,125 @@ def checkNumberOfErrorsByJobId(jobId, errorType='fatal'):
...
+                         job_type_id=JOB_TYPE_DICT['csv_record_validation'],
+                         submission_id=submissionId)
+            sess.add(valJob)
+            sess.flush()
+            # Add dependency between file upload and db upload
+            uploadDependency = JobDependency(job_id=valJob.job_id, prerequisite_id=uploadJob.job_id)
+            sess.add(uploadDependency)
+            jobsRequired.append(valJob.job_id)
+
+        sess.commit()
```
Comment (cmc333333) [inline dataactcore/interfaces/function_bag.py]: It worries me a little bit that not all branches are reaching a `sess.commit()` -- should this be outdented one level?

File: dataactcore/models/sqlalchemy-log-test.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+logging.getLogger('sqlalchemy.engine').setLevel(logging.INFO)
+
+sess = db.dbConnection().session
+job = sess.query(Job).filter(Job.job_id == 1).one()
+
+logger.info('Show data from lazy=joined relationship (job_type): {}'.format(job.job_type.name))
+logger.info('Show data from lazy=None relationship (file_type): {}'.format(job.file_type.name))
```
Comment (cmc333333) [inline dataactcore/models/sqlalchemy-log-test.py]: Should this have been committed?

File: tests/integration/runTests.py [hunk 4]
```diff
@@ -30,16 +30,16 @@ def runTests(argv=''):
...
-    suite.addTests(unittest.makeSuite(JobTests))
-    suite.addTests(unittest.makeSuite(FileTypeTests))
-    suite.addTests(unittest.makeSuite(MixedFileTests))
+    # suite.addTests(unittest.makeSuite(LoginTests))
+    # suite.addTests(unittest.makeSuite(FileTests))
+    # suite.addTests(unittest.makeSuite(UserTests))
+    # suite.addTests(unittest.makeSuite(ValidatorTests))
+    # suite.addTests(unittest.makeSuite(JobTests))
+    # suite.addTests(unittest.makeSuite(FileTypeTests))
+    # suite.addTests(unittest.makeSuite(MixedFileTests))
```
Comment (bsweger) [inline tests/integration/runTests.py]: Ugh, I didn't mean to commit that file at all. Will remove it. Thanks!

File: dataactcore/models/sqlalchemy-log-test.py (hunk 3 above)
Comment (bsweger) [inline dataactcore/models/sqlalchemy-log-test.py]: Nope. Clearly, I was overzealous when adding files. This is from the testing I was doing re: the lazy join 😃

File: dataactcore/interfaces/function_bag.py (hunk 2 above)
Comment (bsweger) [inline dataactcore/interfaces/function_bag.py]: Oops, yes...was definitely the intent to commit everything at the end of the function. Thanks for catching that!

File: dataactcore/interfaces/function_bag.py [hunk 5]
```diff
@@ -118,3 +119,125 @@ def checkNumberOfErrorsByJobId(jobId, errorType='fatal'):
...
+
+    fileTypeId = FILE_TYPE_DICT[fileType]
+
+    # Create a file upload job or, for an existing submission, modify the
+    # existing upload job.
+
+    if existingSubmission:
+        # mark existing upload job as running
+        uploadJob = sess.query(Job).filter(Job.submission_id == submissionId).\
+            filter(Job.file_type_id == fileTypeId, Job.job_type_id == JOB_TYPE_DICT['file_upload']).one()
```
Comment (bsweger) [inline dataactcore/interfaces/function_bag.py]: Well, that is very lovely. Seems to work as expected, so I added it here and in one other place to get in the habit.

File: dataactcore/interfaces/function_bag.py (hunk 1 above)
Comment (bsweger) [inline dataactcore/interfaces/function_bag.py]: Oh! Should we make a story to update this?

**Comment by bsweger** [comment]
@cmc333333 Thanks for the review. Made a few changes based on your feedback!

File: dataactcore/interfaces/function_bag.py (hunk 1 above)
Comment (cmc333333) [inline dataactcore/interfaces/function_bag.py]: :+1: Created DB-1283

**Comment by cmc333333** [comment]
Nice work @bsweger !

**Pull request merged**